lib/dat-worker-pool/default_worker.rb

```diff
@@ -52,14 +45,15 @@ def raise(*args)
     #   If the `ShutdownError` isn't rescued, it will be raised when the worker
     #   is joined.
     def work_loop
-      @on_start_callbacks.each{ |p| p.call(self) }
+      run_callback 'on_start'
       loop do
         break if @shutdown
         fetch_and_do_work
       end
     rescue ShutdownError
     ensure
-      @on_shutdown_callbacks.each{ |p| p.call(self) }
+      run_callback 'on_shutdown'
+      @runner.despawn_worker(self)
```

Comment:
@kellyredding - The default worker now calls despawn directly instead of it being a callback.